plugins/modules/postgresql_info.py

```diff
@@ -761,14 +761,20 @@ def get_ext_info(self):
         ext_dict = {}
         for i in res:
             ext_ver_raw = i[1]
-            ext_ver = i[1].split('.')
+            
+            if re.search('^([0-9]+([\-]*[0-9]+)?\.)+[0-9]+([\-]*[0-9]+)?$', i[1]) == None:
+                ext_ver = [None]
+            else:
+                ext_ver = i[1].split('.')
+                if re.search('-', ext_ver[1]) is None:
```

Comment:
```suggestion
                if re.search('-', ext_ver[1]) is not None:
```
Forget the `not` :confused:

Reply:
Please test it carefully locally:)
Please take a look at `https://github.com/ansible-collections/community.postgresql/tree/main/tests/integration/targets/postgresql_ext/tasks` whether we can put a `postgresql_info` invocation somewhere and check the ret vals. Play with it.
You can run integration tests locally locally before pushing, see the [Quick-start dev guide](https://docs.ansible.com/ansible/devel/community/create_pr_quick_start.html) for details.